src/org/exist/storage/NativeValueIndex.java

```diff
@@ -1323,7 +1323,7 @@ private AtomicValue convertToAtomic( int xpathType, String value )
                 atomic = new StringValue( value ).convertTo( xpathType );
             }
             catch( final XPathException e ) {
-                LOG.warn( "Node value '" + value + "' cannot be converted to " + Type.getTypeName( xpathType ) );
+                LOG.debug( "Node value '" + value + "' cannot be converted to " + Type.getTypeName( xpathType ) );
```

Comment:
One often gets thousands of those messages during a reindex, which potentially hides more severe issues. It is obvious that e.g. an xs:date index can only be applied to dates and the log message doesn't help much to identify the non-compliant elements or attributes within a large data set. It thus has no added value apart from flooding the logs.

If you want to make sure your data is consistent, checking types via a schema is the way to go.